src/app/CommandSender.h

```diff
@@ -181,29 +181,98 @@ class CommandSender final : public Messaging::ExchangeDelegate
         uint16_t remoteMaxPathsPerInvoke = 1;
     };
 
-    // PrepareCommand and FinishCommand are public SDK APIs, so we cannot break source
-    // compatibility for them. By having parameters to those APIs use this struct instead
-    // of individual function arguments, we centralize required changes to one file
-    // when adding new functionality.
-    struct AdditionalCommandParameters
+    // AddRequestData is a public SDK API, so we must maintain source compatibility.
+    // Using this struct for API parameters instead of individual parameters allows us
+    // to make necessary changes for new functionality in a single location.
+    struct AddRequestDataParameters
     {
         // gcc bug requires us to have the constructor below
         // https://gcc.gnu.org/bugzilla/show_bug.cgi?id=96645
-        AdditionalCommandParameters() {}
+        AddRequestDataParameters() {}
 
-        AdditionalCommandParameters & SetStartOrEndDataStruct(bool aStartOrEndDataStruct)
+        AddRequestDataParameters(const Optional<uint16_t> & aTimedInvokeTimeoutMs) : timedInvokeTimeoutMs(aTimedInvokeTimeoutMs) {}
+
+        // When a value is provided for timedInvokeTimeoutMs, this invoke becomes a timed
+        // invoke. CommandSender will use the minimum of all provided timeouts for execution.
+        const Optional<uint16_t> timedInvokeTimeoutMs;
```

Comment:
I am not entirely convinced this needs to be `const` I tried staying true to the original API that passed in this param.